### PR TITLE
docs: fix links

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

This PR fixes the links broken in https://github.com/elastic/apm-server/pull/10894.

A list of broken links is here https://github.com/elastic/apm-server/pull/10891#issuecomment-1563004781.

### CODE REVIEW

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

**Comment by bmorelli25** [comment]
I'm not sure why lint and ci/docs are stuck...

**Comment by vigneshshanmugam** [comment]
@bmorelli25 The CI-docs Action does marks all to success and does not run Lint or other checks. @reakaleek Does it make sense to pass the Lint check as well in the ci-docs action. Its marked as required to pass the PR.

**Comment by reakaleek** [comment]
> @bmorelli25 The CI-docs Action does marks all to success and does not run Lint or other checks. @reakaleek Does it make sense to pass the Lint check as well in the ci-docs action. Its marked as required to pass the PR.

This actually looks a bit confusing. Yes, it would could make sense to create the `Lint` status check in the ci-docs.yml.

However, I think it's weird that it creates the `all` status check, since this one is not even required.

Let me drop a quick PR and adjust the branch protection rules.

**Comment by reakaleek** [comment]
> > @bmorelli25 The CI-docs Action does marks all to success and does not run Lint or other checks. @reakaleek Does it make sense to pass the Lint check as well in the ci-docs action. Its marked as required to pass the PR.
> 
> This actually looks a bit confusing. Yes, it would could make sense to create the `Lint` status check in the ci-docs.yml.
> 
> However, I think it's weird that it creates the `all` status check, since this one is not even required.
> 
> Let me drop a quick PR and adjust the branch protection rules.

https://github.com/elastic/apm-agent-rum-js/pull/1376

**Comment by reakaleek** [comment]
https://github.com/elastic/apm-agent-rum-js/pull/1376 was merged and branch protection rules were adjusted.

**Comment by reakaleek** [comment]
jenkins test this please

File: docs/sourcemap.asciidoc [hunk 1]
```diff
@@ -10,12 +10,12 @@ without losing the ability to quickly and easily debug your applications.
 
 There are three steps required to enable, upload, and apply a source map to error stack traces.
 An overview is listed below, and a complete walk-through is available in the
-{apm-guide-ref}/sourcemaps.html[generate and upload a source map] guide.
+{apm-guide-ref}/source-map-how-to.html[generate and upload a source map] guide.
 
 1. Set the <<service-version,`serviceVersion`>> when initializing the RUM Agent.
-2. {apm-guide-ref}/sourcemaps.html#sourcemap-rum-generate[Generate a source map]
+2. {apm-guide-ref}/source-map-how-to.html#sourcemap-rum-generate[Generate a source map]
```
Comment (bmorelli25) [inline docs/sourcemap.asciidoc:16]:
```suggestion
2. {apm-guide-ref}/source-map-how-to.html#source-map-rum-generate[Generate a source map]
```

File: docs/sourcemap.asciidoc [hunk 2]
```diff
@@ -10,12 +10,12 @@ without losing the ability to quickly and easily debug your applications.
...
 for your application with the `serviceVersion` from step one.
-3. {apm-guide-ref}/sourcemaps.html#sourcemap-rum-upload[Enable and upload the source map file] to APM Server.
+3. {apm-guide-ref}/source-map-how-to.html#sourcemap-rum-upload[Enable and upload the source map file] to APM Server.
```
Comment (bmorelli25) [inline docs/sourcemap.asciidoc:18]:
```suggestion
3. {apm-guide-ref}/source-map-how-to.html#source-map-rum-upload[Enable and upload the source map file] to APM Server.
```

**Comment by bmorelli25** [comment]
@vigneshshanmugam can you merge this PR for me? I don't have permission.

**Pull request merged**